utils/fork-tree/src/lib.rs

```diff
@@ -267,6 +188,53 @@ where
 		self.node_iter().map(|node| (&node.hash, &node.number, &node.data))
 	}
 
+	/// Map fork tree into values of new types.
+	///
+	/// Tree traversal technique (e.g. BFS vs DFS) is left as not specified and
+	/// may be subject to change in the future. In other words, your predicates
+	/// should not rely on the observed traversal technique currently in use.
+	pub fn map<VT, F>(self, f: &mut F) -> ForkTree<H, N, VT>
+	where
+		F: FnMut(&H, &N, V) -> VT,
+	{
+		let mut queue: Vec<_> =
+			self.roots.into_iter().rev().map(|node| (usize::MAX, node)).collect();
```

Comment:
This is nitpicking but I would have used an option here instead of `usize::MAX`.

Reply:
I doubt it matters here, although please note that using an `Option` here will bloat each element from 8 bytes (the size of `usize`) to 16 bytes (the size of `Option<usize>`). (: